### PR TITLE
Feature/update search filter to be like GitHub

### PR DESCRIPTION
**Pull request by fpernin**
#### :heavy_check_mark: Checklist

- [x] Added tests for new functionality and regression tests for bug fixes
- [x] Screenshots of before and after attached (for UI changes)
- [x] Added or updated documentation (if applicable)

github-pull-request

### CODE REVIEW

File: plugins/frontend/backstage-plugin-github-pull-requests/src/api/GithubPullRequestsApi.ts [hunk 1]
```diff
@@ -15,34 +15,36 @@
...
     owner: string;
     repo: string;
+    defaultFilter: string;
     pageSize?: number;
     page?: number;
     branch?: string;
     state?: PullRequestState;
     baseUrl: string|undefined;
   }) => Promise<{
-    maxTotalItems?: number;
```
Comment (jpatovh) [inline plugins/frontend/backstage-plugin-github-pull-requests/src/api/GithubPullRequestsApi.ts:45]: not used anymore?

File: plugins/frontend/backstage-plugin-github-pull-requests/src/api/GithubPullRequestsClient.ts [hunk 2]
```diff
@@ -16,48 +16,43 @@
...
       page,
-      owner,
     });
-    const paginationLinks = pullRequestResponse.headers.link;
-    const lastPage = paginationLinks?.match(/\d+(?!.*page=\d*)/g) || ['1'];
-    const maxTotalItems = paginationLinks?.endsWith('rel="last"')
-      ? parseInt(lastPage[0], 10) * pageSize
-      : undefined;
     return {
-      maxTotalItems,
```
Comment (jpatovh) [inline plugins/frontend/backstage-plugin-github-pull-requests/src/api/GithubPullRequestsClient.ts:59]: not used anymore?

File: plugins/frontend/backstage-plugin-github-pull-requests/src/components/PullRequestsTable/PullRequestsTable.tsx [hunk 3]
```diff
@@ -14,10 +14,13 @@
...
 import React, { FC, useState } from 'react';
-import { Typography, Box, ButtonGroup, Button } from '@material-ui/core';
+import { InputAdornment, IconButton, TextField, Typography, Box, ButtonGroup, Button } from '@material-ui/core';
 import GitHubIcon from '@material-ui/icons/GitHub';
+import ClearIcon from '@material-ui/icons/Clear';
+
+import SearchIcon from '@material-ui/icons/Search';
 import { Table, TableColumn, MissingAnnotationEmptyState } from '@backstage/core-components';
-import { isGithubSlugSet, GITHUB_PULL_REQUESTS_ANNOTATION } from '../../utils/isGithubSlugSet';
+import { isGithubSlugSet, GITHUB_PULL_REQUESTS_ANNOTATION, isRoadieBackstageDefaultFilter } from '../../utils/isGithubSlugSet';
```
Comment (jpatovh) [inline plugins/frontend/backstage-plugin-github-pull-requests/src/components/PullRequestsTable/PullRequestsTable.tsx]: to respect "isGithubSlugSet", please put "isRoadieBackstageDefaultFilterSet"

File: plugins/frontend/backstage-plugin-github-pull-requests/src/utils/isGithubSlugSet.ts [hunk 4]
```diff
@@ -17,7 +17,11 @@
...
 
 export const GITHUB_PULL_REQUESTS_ANNOTATION = 'github.com/project-slug';
+export const ROADIE_BACKSTAGE_PULL_REQUESTS_DEFAULT_FILTER_ANNOTATION = 'roadie-backstage-pull-requests/default-filter';
 
 export const isGithubSlugSet = (entity: Entity) => {
   return entity?.metadata.annotations?.[GITHUB_PULL_REQUESTS_ANNOTATION] ?? '';
 };
+export const isRoadieBackstageDefaultFilter = (entity: Entity) => {
+  return entity?.metadata.annotations?.[ROADIE_BACKSTAGE_PULL_REQUESTS_DEFAULT_FILTER_ANNOTATION] ?? '';
+};
```
Comment (jpatovh) [inline plugins/frontend/backstage-plugin-github-pull-requests/src/utils/isGithubSlugSet.ts]: I think it's better to put it in different file, like src/utils/isRoadieBackstageDefaultFilterSet.ts

File: plugins/frontend/backstage-plugin-github-pull-requests/src/utils/isGithubSlugSet.ts [hunk 5]
```diff
@@ -17,7 +17,11 @@
 import { Entity } from '@backstage/catalog-model';
 
 export const GITHUB_PULL_REQUESTS_ANNOTATION = 'github.com/project-slug';
+export const ROADIE_BACKSTAGE_PULL_REQUESTS_DEFAULT_FILTER_ANNOTATION = 'roadie-backstage-pull-requests/default-filter';
```
Comment (jpatovh) [inline plugins/frontend/backstage-plugin-github-pull-requests/src/utils/isGithubSlugSet.ts]: you've to update the README.md, to explain this new feature, too :)

**Review by jpatovh — CHANGES_REQUESTED** [review]

File: plugins/frontend/backstage-plugin-github-pull-requests/src/api/GithubPullRequestsApi.ts (hunk 1 above)
Comment (fpernin) [reply plugins/frontend/backstage-plugin-github-pull-requests/src/api/GithubPullRequestsApi.ts:45]: yes it was a bad value calculate by the api call function it was page multiply by item per page and in the new endpoint to have pull request we have the real total value

File: plugins/frontend/backstage-plugin-github-pull-requests/src/api/GithubPullRequestsClient.ts (hunk 2 above)
Comment (fpernin) [reply plugins/frontend/backstage-plugin-github-pull-requests/src/api/GithubPullRequestsClient.ts:59]: yes it was a bad value calculate by the api call function it was page multiply by item per page and in the new endpoint to have pull request we have the real total value

File: plugins/frontend/backstage-plugin-github-pull-requests/src/components/PullRequestsStatsCard/PullRequestsStatsCard.test.tsx [hunk 6]
```diff
@@ -50,7 +50,7 @@ describe('PullRequestsCard', () => {
   beforeEach(() => {
```
Comment (pierrickmolera) [inline plugins/frontend/backstage-plugin-github-pull-requests/src/components/PullRequestsStatsCard/PullRequestsStatsCard.test.tsx:50]: I see previous lines about config, ApiRegistry are copied in each test file ... it was done like that before. what do you think about a common file ?

**Review by pierrickmolera — COMMENTED** [review]
Good for me

**Pull request closed without merging**